old/rootfs_provider/docker_rootfs_provider.go

```diff
@@ -55,7 +58,17 @@ func (provider *dockerRootFSProvider) ProvideRootFS(logger lager.Logger, id stri
 
 	imageID, envvars, volumes, err := provider.repoFetcher.Fetch(logger, url, tag)
 	if err != nil {
-		return "", nil, err
+		if url.Host == "" {
```

Comment:
rather than wrapping the error in an if/else it would be good if the original error was more descriptive and especially if the determination of what remote was used was only made in one place. Tthe [repositoryProvider](https://github.com/cloudfoundry-incubator/garden-linux/blob/7c655827fe0e4134f44f87cbbced98bcab5137f6/old/repository_fetcher/repository_provider.go#L31) and [repositoryFetcher](https://github.com/cloudfoundry-incubator/garden-linux/blob/7c655827fe0e4134f44f87cbbced98bcab5137f6/old/repository_fetcher/repository_fetcher.go#L93) are the objects that know what they're going to connect to if url.Host is empty, so it'd be best not to duplicate the logic around choosing a default in that case here (i.e. we should avoid having multiple places where we 'know' we use registry.IndexServerAddress if Host is empty).